File: javascript/1-TwoSum.js
// Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

// You may assume that each input would have exactly one solution, and you may not use the same element twice.

// You can return the answer in any order.
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    let indexArray = [];

    for(let i = 0; i < nums.length; i++) {
        for(let j = i+1; j < nums.length; j++) {
            if(nums[i] + nums[j] == target) {
                indexArray.push(i);
                indexArray.push(j);
                break;
            }
        }
    }

    return indexArray;
};

console.log(twoSum([2,7,11,15], 9));
console.log(twoSum([3,2,4], 6));
console.log(twoSum([3,3], 6));

File: java/1-twoSum.java
import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        int[] res = new int[2];
        for(int i = 0; i < nums.length; i++) {
            int n = nums[i];
            if(map.containsKey(target - n)) {
                res[0] = i;
                res[1] = map.get(target - n);
                break;
            }
            map.put(n, i);
        }

        return res;
    }
}